.github/workflows/e2e-tests.yml

```
@@ -12,20 +12,86 @@ on:
         type: string
 
 jobs:
+
   e2e:
-    name: "E2E Tests"
+    name: E2E Tests - Shard ${{ matrix.shard }} of ${{ matrix.total_shards }}
```

nit: can refine content a little
```suggestion
    name: Run E2E tests (${{ matrix.shard }} of ${{ matrix.total_shards }})
```